presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnChecksum.java

```diff
@@ -63,19 +101,44 @@ public boolean equals(Object obj)
         }
         ArrayColumnChecksum o = (ArrayColumnChecksum) obj;
         return Objects.equals(checksum, o.checksum) &&
+                Objects.equals(floatingPointChecksum, o.floatingPointChecksum) &&
                 Objects.equals(cardinalityChecksum, o.cardinalityChecksum) &&
                 Objects.equals(cardinalitySum, o.cardinalitySum);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(checksum, cardinalityChecksum, cardinalitySum);
+        return Objects.hash(checksum, floatingPointChecksum, cardinalityChecksum, cardinalitySum);
     }
 
     @Override
     public String toString()
     {
-        return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        if (Objects.isNull(floatingPointChecksum.getSum())) {
+            return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        }
+        else {
+            return format(
+                    "%s, cardinality_checksum: %s, cardinality_sum: %s",
+                    floatingPointChecksum.toString(),
+                    cardinalityChecksum,
+                    cardinalitySum);
+        }
+    }
+
+    public static FloatingPointColumnChecksum.ValidateResult validateFloatingPoint(
+            ArrayColumnChecksum controlChecksum,
+            ArrayColumnChecksum testChecksum,
+            double relativeErrorMargin,
+            double absoluteErrorMargin)
+    {
+        // Check the non floating point members first.
+        if (!Objects.equals(controlChecksum.getCardinalityChecksum(), testChecksum.getCardinalityChecksum()) ||
+                !Objects.equals(controlChecksum.getCardinalitySum(), testChecksum.getCardinalitySum())) {
+            return new FloatingPointColumnChecksum.ValidateResult(false, Optional.of("cardinality mismatch"));
```

Comment:
Do we need this new class? Can we use ColumnMatchResult instead?

Reply:
ColumnMatchResult is a templated class and we must return different instantiations of it from ArrayColumnChecksum and FloatingPointColumnChecksum.

I decide to have that small class to carry actual validation result, except all the columns info, which will be provided by concrete validators.

Reply:
The `ValidateResult` still feel not needed, I don't see particular bad effect of using `ColumnMatchResult`

Reply:
Ok, I guess, we can get rid of it.